engine.js

```diff
@@ -679,7 +679,29 @@ var resolveError = function(err) {
     };
 
     HMDAEngine.isValidNumGinnieMaeVALoans = function(hmdaFile) {
-        return true;
+        var numGinnieLoans = 0,
+            numLoans = 0;
+        _.each(hmdaFile.loanApplicationRegisters, function(element, index, next) {
+            if (_.contains(['1', '3'], element.loanPurpose) && _.contains(['1', '6'], element.actionTaken) &&
+                _.contains(['1', '2'], element.propertyType) && (element.loanType === '3')) {
+                numLoans++;
+                if (element.purchaserType === '2') {
+                    numGinnieLoans ++;
+                }
+            }
+        });
+        return this.apiGET('isValidNumLoans/ginnieMae', [hmdaFile.transmittalSheet.respondentID, numLoans, numGinnieLoans])
+        .then(function(body) {
+            var result = resultFromResponse(body);
+            if (result.result) {
+                return true;
+            } else {
+                result.currentLoans = numLoans;
+                result.currentGinnieLoans = numGinnieLoans;
+                var error = [{'properties': result}];
```

Comment:
Now that you've updated the return from the API in cfpb/hmda-edit-check-api#43, you no longer need to include `result.currentLoans` or `result.currentGinnieLoans`. Also, you will need to delete `result.result` before you assign `result` to the `properties` object.